verilog/analysis/checkers/suggest_parentheses_rule_test.cc

```diff
@@ -0,0 +1,44 @@
+// Copyright 2017-2020 The Verible Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "verilog/analysis/checkers/suggest_parentheses_rule.h"
+
+
+#include "common/analysis/syntax_tree_linter_test_utils.h"
+#include "verilog/analysis/verilog_analyzer.h"
+
+namespace verilog {
+namespace analysis {
+namespace {
+
+using verible::LintTestCase;
+using verible::RunLintTestCases;
+
+TEST(SuggestParenthesesTest, AcceptTests) {
+  const std::initializer_list<LintTestCase> kSuggestParenthesesTestCases = {
+      {""},
+      {"module m;\nendmodule\n"},
+      {"module m;\ninitial begin end\nendmodule"},
+      {"module m;\n", "assign foo = condition_a? a : b;\n", "endmodule"},
+      {"module m;\n", "assign foo = condition_a? (condition_b? a : b) : c;\n", "endmodule"},
+      {"module m;\n", "assign foo = condition_a? (condition_b? a : b) : (condition_c? c : d);\n", "endmodule"},
+      {"module m;\n", "assign foo = condition_a? (condition_b? (condition_c? a : b) : c) : d;\n", "endmodule"},
+  };
```

Comment:
Add some test cases that expect to find violations like this:

```c++
constexpr int kTag = 1;  // don't care
const std::initializer_list<LintTestCase> kSuggestParenthesesTestCases = {
  ...
  {"module m;\nassign foo = condition_a? ", {kTag, "condition_b? a : b"}, " : c;\nendmodule"},
};
```

This notation `{kTag, "condition_b? a : b"}` says: "expect to find a violation on this exact substring range."

Reply:
Truth to be said, I was going to ask about this part and how to do that. Thanks for this info. :)

Reply:
Glad that helped!  Don't forget to "request review" when ready.  Thanks for working on this again.